### ISSUE TITLE
Add staggered PSATD and PSTD solver

### ISSUE THREAD

**Issue by MKirchen** [fbpic/fbpic#17]
I think at some point it would make sense to add a staggered version of the (Galilean)-PSATD as well as the PSTD solver. This will enable us to make more comparisons between the different solvers in the future.

@RemiLehe What do you think?

**Comment by RemiLehe** [comment]
Yes, this would be interesting to have in the code. 
Just to be sure: by staggered, do you mean staggered in time ?

**Comment by MKirchen** [comment]
Yes I meant only staggered in time. I think it would not make too much sense to add it in space or do you think this would be a valuable option?

**Comment by RemiLehe** [comment]
No, I agree that we should do staggering only in time, as it is much simpler.
